numpy/random/tests/test_random.py

```diff
@@ -400,6 +400,7 @@ def test_choice_exceptions(self):
         assert_raises(ValueError, sample, [1, 2], 3, p=[1.1, -0.1])
         assert_raises(ValueError, sample, [1, 2], 3, p=[0.4, 0.4])
         assert_raises(ValueError, sample, [1, 2, 3], 4, replace=False)
+        assert_raises(ValueError, sample, [1, 2, 3], -2, replace=False)
```

Comment:
Can you test with `(-1,)` and `(-1, 1)` too?

Reply:
@eric-wieser You mean `size=(-1,1)`? Shouldn't it be a 1D array (is that what we're checking here)?

Reply:
Yes, I do mean that. The first argument is a 1d array, but the output shape can be any dimension. Your new tests look good, although you might want to reference the issue number as `# gh-NNNNN` in a comment

Reply:
Added `# gh-<issueno>`
@eric-wieser Thanks for patiently reviewing this PR! Is there anything more?